### ISSUE TITLE
File without formatter causes other files to not be formatted

### ISSUE THREAD

**Issue by Ma27** [numtide/treefmt#441]
**Describe the bug**

When passing files to the treefmt cli (e.g. as part of a pre-commit hook), subsequent files don't get formatted if a file before doesn't have a formatter configured.

**To Reproduce**

* Given two files `pyproject.toml` (no formatter) and `foo/bar.py` (ruff)
* Given the following treefmt-nix configuration (all in a flake.nix using the flake-parts modules):

  ```nix
  {
    projectRootFile = "flake.nix";
    programs = {
      nixfmt.enable = true;
      ruff.enable = true;
    };
  }
  ```
* and the following git-hooks.nix configuration
  ```nix
  { pre-commit.settings.hooks.treefmt.enable = true; }
  ```

* Make changes to `pyproject.toml` & `foo/bar.py` and stage them. The change in `foo/bar.py` must cause a reformat in ruff.
* When committing, the pre-commit hook won't complain.

**Expected behavior**

The pre-commit hook should complain.

**System information**

NixOS 24.05, treefmt 2.0.5 (via treefmt-nix 879b29ae9a0378904fbbefe0dadaed43c8905754), git-hooks.nix at rev 85f7a7177c678de68224af3402ab8ee1bcee25c8.

**Additional context**

I tracked it down to the following two CLI invocations:

The following rightfully fails because `foo/bar.py` requires a reformat:
```
$ /nix/store/frmyslc7ayg3afpcsnd5whn0zrhsrs6k-treefmt-2.0.5/bin/treefmt --config-file=/nix/store/albr8h04ghzrp8wdcs0xbziz9mg37xm2-treefmt.toml --tree-root-file=flake.nix --fail-on-change foo/bar.py flake.nix   -vvv
```

The following wrongly doesn't complain because `pyproject.toml` doesn't have a formatter:

```
$ /nix/store/frmyslc7ayg3afpcsnd5whn0zrhsrs6k-treefmt-2.0.5/bin/treefmt --config-file=/nix/store/albr8h04ghzrp8wdcs0xbziz9mg37xm2-treefmt.toml --tree-root-file=flake.nix --fail-on-change pyproject.toml foo/bar.py flake.nix
traversed 1 files
emitted 0 files for processing
formatted 0 files (0 changed) in 0s

```

If `pyproject.toml` is the last file in the CLI (rather then the first), treefmt correctly complains about foo/bar.py.

The bug is also reproducible after an `rm -r ~/.cache/treefmt/`.

**Comment by brianmcgee** [comment]
I've been working on how path arguments are handled and refactoring the way we walk the filesystem, most notably:

- https://github.com/numtide/treefmt/pull/436
- https://github.com/numtide/treefmt/pull/438

Are you able to quickly test with the latest main?

**Comment by Ma27** [comment]
@brianmcgee thanks for the pointer! Yes, this is indeed fixed on master (tested 659aa0f).

**Comment by brianmcgee** [comment]
That's handy :smile: 

I'll be cutting a release in the next week after I get a few other changes in.

**Comment by brianmcgee** [comment]
Closed by #438

**Comment by Ma27** [comment]
Splendid, thanks!
